src/lightning_lite/cli.py

```diff
@@ -0,0 +1,177 @@
+        default="127.0.0.1",
+        help="The hostname or IP address of the main machine (usually the one with node_rank = 0).",
+    )
+    parser.add_argument(
+        "--main-port",
+        type=int,
+        default=29400,
+        help="The main port to connect to the main machine.",
+    )
+    parser.add_argument(
+        "--precision",
+        type=str,
+        default="32",
+        choices=_SUPPORTED_PRECISION,
+        help=(
+            "Double precision (``64``), full precision (``32``), half precision (``16``) or bfloat16 precision"
+            " (``'bf16'``)"
+        ),
+    )
+
+    args, script_args = parser.parse_known_args()
+    return args, script_args
+
+
+def _set_env_variables(args: Namespace) -> None:
+    """Set the environment variables for the new processes.
+
+    The Lite connector will parse the arguments set here.
+    """
+    os.environ["LT_CLI_USED"] = "1"
+    os.environ["LT_ACCELERATOR"] = str(args.accelerator)
+    if args.strategy is not None:
+        os.environ["LT_STRATEGY"] = str(args.strategy)
+    os.environ["LT_DEVICES"] = str(args.devices)
+    os.environ["LT_NUM_NODES"] = str(args.num_nodes)
+    os.environ["LT_PRECISION"] = str(args.precision)
+
+
+def _get_num_processes(accelerator: str, devices: str) -> int:
+    """Parse the `devices` argument to determine how many processes need to be launched on the current machine."""
+    if accelerator == "gpu":
+        parsed_devices = _parse_gpu_ids(devices, include_cuda=True, include_mps=True)
+    elif accelerator == "cuda":
+        parsed_devices = CUDAAccelerator.parse_devices(devices)
+    elif accelerator == "mps":
+        parsed_devices = MPSAccelerator.parse_devices(devices)
+    elif accelerator == "tpu":
+        raise ValueError("Launching processes for TPU through the CLI is not supported.")
+    else:
+        return CPUAccelerator.parse_devices(devices)
+    return len(parsed_devices) if parsed_devices is not None else 0
+
+
+def _torchrun_launch(args: Namespace, script_args: List[str]) -> None:
+    """This will invoke `torchrun` programmatically to launch the given script in new processes."""
+
+    if _IS_WINDOWS and _TORCH_GREATER_EQUAL_1_13:
+        # TODO: remove once import issue is resolved: https://github.com/pytorch/pytorch/issues/85427
+        _log.error(
+            "On the Windows platform, this launcher is currently only supported on torch < 1.13 due to a bug"
+            " upstream: https://github.com/pytorch/pytorch/issues/85427"
+        )
+        exit(1)
+
+    if args.strategy == "dp":
+        num_processes = 1
+    else:
+        num_processes = _get_num_processes(args.accelerator, args.devices)
+
+    torchrun_args = [
+        "--nproc_per_node",
+        str(num_processes),
+        "--nnodes",
+        str(args.num_nodes),
+        "--node_rank",
+        str(args.node_rank),
+        "--master_addr",
+        args.main_address,
+        "--master_port",
+        str(args.main_port),
```

Comment:
nit: this is more readable

```suggestion
        f"--nproc_per_node={num_processes}",
        f"--nnodes={args.num_nodes}",
        f"--node_rank={args.node_rank}",
        f"--master_addr={args.main_address}",
        f"--master_port={args.main_port}",
```